network-api/networkapi/wagtailpages/pagemodels/profiles.py

```diff
@@ -63,8 +65,9 @@ def __str__(self):
         return self.name
 
     def save(self, *args, **kwargs):
-        self.slug = slugify(f"{self.name}-{str(self.id)}")
+        self.slug = slugify(self.name)
         super(Profile, self).save(*args, **kwargs)
+        self._meta.model.objects.filter(id=self.id).update(slug=Concat(F("slug"), Value("-"), F("id")))
```

Comment:
I'm a tad confused as to using ids/pks here considering https://github.com/MozillaFoundation/foundation.mozilla.org/issues/10413 but I've approved this given your comments on that issue... looks like you are aware of the surrounding issues

Reply:
The issue in #10413 is to remove the `id` from the URL in the author detail route (the route takes both the slug and the id). I think we still need the `id` on the slug to make sure that it is unique, although we're gonna need to confirm this once we get to implementing #10413